packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,363 @@
+        PlatformVersion: this.options?.platformVersion,
+      },
+    };
+  }
+}
+
+/**
+ * Configuration for running an ECS task on EC2
+ *
+ * @see https://docs.aws.amazon.com/AmazonECS/latest/userguide/launch_types.html#launch-type-ec2
+ */
+export class EcsEc2LaunchTarget implements IEcsLaunchTarget {
+  constructor(private readonly options?: EcsEc2LaunchTargetOptions) {}
+  /**
+   * Called when the EC2 launch type is configured on RunTask
+   */
+  public bind(_task: EcsRunTask, taskDefinition: ecs.ITaskDefinition, cluster?: ecs.ICluster): EcsLaunchTargetConfig {
+    if (!taskDefinition.isEc2Compatible) {
+      throw new Error('Supplied TaskDefinition is not compatible with Fargate');
+    }
+
+    if (!cluster?.hasEc2Capacity) {
+      throw new Error('Cluster for this service needs Ec2 capacity. Call addXxxCapacity() on the cluster.');
+    }
+
+    return {
+      parameters: {
+        LaunchType: 'EC2',
+        PlacementConstraints: noEmpty(flatten((this.options?.placementConstraints ?? []).map((c) => c.toJson().map(uppercaseKeys)))),
+        PlacementStrategy: noEmpty(flatten((this.options?.placementStrategies ?? []).map((c) => c.toJson().map(uppercaseKeys)))),
+      },
+    };
+
+    function uppercaseKeys(obj: { [key: string]: any }): { [key: string]: any } {
+      const ret: { [key: string]: any } = {};
+      for (const key of Object.keys(obj)) {
+        ret[key.slice(0, 1).toUpperCase() + key.slice(1)] = obj[key];
+      }
+      return ret;
+    }
+
+    function flatten<A>(xs: A[][]): A[] {
+      return Array.prototype.concat([], ...xs);
+    }
+
+    function noEmpty<A>(xs: A[]): A[] | undefined {
+      if (xs.length === 0) {
+        return undefined;
+      }
+      return xs;
+    }
+  }
+}
+
+/**
+ * Run a Task on ECS or Fargate
+ */
+export class EcsRunTask extends sfn.TaskStateBase implements ec2.IConnectable {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  /**
+   * Manage allowed network traffic for this service
+   */
+  public readonly connections: ec2.Connections = new ec2.Connections();
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private securityGroup?: ec2.ISecurityGroup;
+  private networkConfiguration?: any;
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: EcsRunTaskProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EcsRunTask.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN && !sfn.FieldUtils.containsTaskToken(props.containerOverrides)) {
```

Comment:
can we use `JsonPath` instead of `FieldUtils`?

Reply:
I think that's a good idea, but we can't yet.
Currently, `JsonPath` will allow you to retrieve the task token field, but should probably add the `containsTaskToken` method for convenience